src/Symfony/Component/Serializer/Exception/MissingConstructorArgumentsException.php

```diff
@@ -17,19 +17,52 @@
 class MissingConstructorArgumentsException extends RuntimeException
 {
     /**
+     * @deprecated since Symfony 6.3
+     *
      * @var string[]
      */
     private $missingArguments;
 
-    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [])
+    private ?string $class;
+    private ?string $missingArgument;
+
+    public function __construct(string $message, int $code = 0, \Throwable $previous = null, /* string */ $class = null, string $missingArgument = null)
     {
-        $this->missingArguments = $missingArguments;
+        if (\is_array($class)) {
+            trigger_deprecation('symfony/serializer', '6.3', 'Passing an array of missing arguments as the fourth argument of the "%s" constructor is deprecated, pass the missing arguments class name instead and the missing arguments name as the fifth argument.', __CLASS__);
+
+            $this->missingArguments = $class;
+
+            $class = null;
+            $missingArgument = $this->missingArguments[0] ?? null;
+        }
+
+        if (null !== $class && !\is_string($class)) {
+            throw new \TypeError(sprintf('Argument 4 passed to "%s()" must be a string or null, "%s" given.', __METHOD__, get_debug_type($class)));
+        }
+
+        $this->class = $class;
+        $this->missingArgument = $missingArgument;
+
+        $this->missingArguments ??= null !== $this->missingArgument ? [$this->missingArgument] : [];
 
         parent::__construct($message, $code, $previous);
     }
 
+    public function getClass(): ?string
+    {
+        return $this->class;
+    }
+
+    public function getMissingArgument(): ?string
```

Comment:
```suggestion
    public function getMissingConstructorArgument(): ?string
```
?

Reply:
I ok either way, it just seemed kind of redundant since the exception is called `MissingConstructorArgumentsException`.

Reply:
A class supporting a single argument should rather be named `MissingConstructorArgumentException` than `MissingConstructorArgumentsException`

Reply:
I've added a new `MissingConstructorArgumentException` class and deprecated the old one, however, since the main goal of this PR is to add the class to the exception, I can revert the renaming part if you think the change is not worth it.